src/lib/stores/api.js

```diff
@@ -80,8 +82,27 @@ async function getStudies() {
 }
 
 const _stateChangeCallbacks = [];
-const _authChangeCallbacks = [];
-const _connectedChangeCallbacks = [];
+const _connectedChangeCallbacks = [async (/** @type {string} */ studyId, /** @type {{ [utmCode: string]: string; }} */ attribution) => {
+  analytics = (await import("firebase/analytics")).getAnalytics();
+  logEvent = (await import("firebase/analytics")).logEvent;
+
+  // TODO figure out how to send attribution in a way FA will use it automatically.
+  const eventParams = { studyId };
+  ["source", "medium", "campaign", "term", "content"].forEach(code => {
+    if (code in attribution) {
+      eventParams[code] = attribution[code];
+    }
+  })
+  logEvent(analytics, "activate_extension", eventParams);
+}];
+
+const _authChangeCallbacks = [async (/** @type {import("firebase/auth").User} */ user) => {
+  analytics = (await import("firebase/analytics")).getAnalytics();
+  logEvent = (await import("firebase/analytics")).logEvent;
+
+  const loggedIn = Boolean(user && user.uid);
+  logEvent(analytics, `sign_${loggedIn ? "in" : "out"}`);
```

Comment:
There is a login standard event. Not sure if we need to log this. It should log it for both email and google sign in.

Reply:
Ya I suspect we already do get those standard events (I will check).

Reply:
Ok it seems we need to log these but the recommended name is "login" and I suspect it ties into other things so maybe we can change the name here. Here is the link I found: https://developers.google.com/android/reference/com/google/firebase/analytics/FirebaseAnalytics.Event#LOGIN

Reply:
Here is a list of automatically logged events FYI: https://support.google.com/analytics/answer/9234069